lib/hyrax/specs/shared_specs/indexers.rb

```diff
@@ -72,26 +121,40 @@
     }
   end
 
-  let(:resource_class) do
-    Class.new(Hyrax::Work) do
-      include Hyrax::Schema(:basic_metadata)
-    end
-  end
-
   describe '#to_solr' do
     it 'indexes basic metadata' do
       expect(indexer.to_solr)
-        .to include(keyword_sim: a_collection_containing_exactly(*attributes[:keyword]),
+        .to include(keyword_sim:   a_collection_containing_exactly(*attributes[:keyword]),
                     subject_tesim: a_collection_containing_exactly(*attributes[:subject]),
                     subject_sim:   a_collection_containing_exactly(*attributes[:subject]))
     end
   end
 end
 
+RSpec.shared_examples 'a Work indexer' do
+  before do
+    raise 'indexer_class must be set with `let(:indexer_class)`' unless defined? indexer_class
+    raise 'resource must be set with `let(:resource)` and is expected to be a kind of Hyrax::Work' unless defined?(resource) && resource.kind_of?(Hyrax::Work)
+    # optionally can pass in default_visibility by setting it with a let statement if your application changes the default; Hyrax defines this as 'restricted'
+    # See samvera/hyrda-head hydra-access-controls/app/models/concerns/hydra/access_controls/access_rights.rb for possible VISIBILITY_TEXT_VALUE_...'
+  end
+  subject(:indexer) { indexer_class.new(resource: resource) }
+
+  it_behaves_like 'a Hyrax::Resource indexer'
+  it_behaves_like 'a Core metadata indexer'
+  it_behaves_like 'a permission indexer'
+  it_behaves_like 'a visibility indexer'
+end
+
 RSpec.shared_examples 'a Collection indexer' do
+  before do
+    raise 'indexer_class must be set with `let(:indexer_class)`' unless defined? indexer_class
+    raise 'resource must be set with `let(:resource)` and is expected to be a kind of Hyrax::PcdmCollection' unless defined?(resource) && resource.kind_of?(Hyrax::PcdmCollection)
+  end
   subject(:indexer) { indexer_class.new(resource: resource) }
-  let(:resource)    { Hyrax::PcdmCollection.new }
 
+  it_behaves_like 'a Hyrax::Resource indexer'
+  it_behaves_like 'a Core metadata indexer'
```

Comment:
The shared spec defines the minimal requirements for a collection instead of expecting the  individual collection tests to know which parts are required.